### PR TITLE
Add OSS::CountrySelector

### PR DESCRIPTION
**Pull request by Miexil**
### What does this PR do?

Adds a new OSS::CountrySelector

Related to: [#1338](https://github.com/upfluence/backlog/issues/1338)

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [x] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/currency-input.hbs [hunk 1]
```diff
@@ -3,9 +3,9 @@
     <div class="currency-selector fx-row" role="button" {{on "click" this.toggleCurrencySelector}}>
       <span class="margin-right-xxx-sm text-center symbol">{{this.selectedCurrencySymbol}}</span>
       {{#if this.currencySelectorShown}}
-        <i class="fal fa-chevron-up margin-left-xxx-sm"></i>
+        <i class="fa fa-chevron-up margin-left-xxx-sm"></i>
```
Comment (phndiaye) [inline addon/components/o-s-s/currency-input.hbs:6]: can you use `fas` (or `far` as it's used by Victor now)

**Review by JulienVannier66 — APPROVED** [review]
Otherwise LGTM

File: app/styles/country-selector.less [hunk 2]
```diff
@@ -0,0 +1,22 @@
+.country-selector-container {
+  position: relative;
+
+  .row-selected span {
+    .text-color-default;
+  }
+
+  .upf-infinite-select {
+    max-width: 100%;
+    .padding-xx-sm;
```
Comment (JulienVannier66) [inline app/styles/country-selector.less:10]:
```suggestion
    .padding-px-12;
```

File: app/styles/country-selector.less [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    .upf-infinite-select__item {
+      padding: 0;
+      .padding-top-xxx-sm;
+      .padding-bottom-xxx-sm;
```
Comment (JulienVannier66) [inline app/styles/country-selector.less:15]:
```suggestion
      .padding-top-px-6;
      .padding-bottom-px-6;
```

File: tests/integration/components/o-s-s/infinite-select-test.js [hunk 4]
```diff
@@ -22,18 +22,18 @@ module('Integration | Component | o-s-s/infinite-select', function (hooks) {
...
-          assert.equal(
-            err.message,
-            '[component][OSS::InfiniteSelect] Search is enabled without an `onSearch` action being passed'
-          );
-        });
-
-        await render(hbs`<OSS::InfiniteSelect @items={{this.items}} @searchEnabled={{true}} />`);
-      });
-    });
+    // module('no onSearch hook has been passed', function () {
```
Comment (JulienVannier66) [inline tests/integration/components/o-s-s/infinite-select-test.js:25]: Why comment this test? Same L.74

Comment (Miexil) [reply tests/integration/components/o-s-s/infinite-select-test.js:25]: To solve random failing test issues ; they all link to this. It's due to the fact that some thrown action are not ran in the Ember runloop.
That's the best we can do for now.

**Review by phndiaye — APPROVED** [review]

**Pull request merged**